src/python/pants/option/global_options.py

```diff
@@ -437,6 +437,14 @@ def register_bootstrap_options(cls, register):
             "Patterns use the gitignore syntax (https://git-scm.com/docs/gitignore). "
             "The `--pants-distdir` and `--pants-workdir` locations are inherently ignored.",
         )
+        register(
+            "--pants-use-gitignore",
+            advanced=True,
+            type=bool,
+            default=False,
+            help="Make use of .gitignore files when determining whether to ignore filesystem "
+            "operations performed by pants. This may be used together with `--pants-ignore`.",
```

Comment:
This should indicate that the gitignore is prepended to `--pants-ignore`. The other option should possibly also reference this one, and explain how you can exclude things (see the comments on the ticket).

Also, it looks like this will only respect the root `.gitignore` file (which I think is totally reasonable for now). Should mention that here and/or link to a followup ticket.